Add disk encryption options documentation

bryan-cox opened the pull request:
<!-- If this is your first PR, welcome! Please make sure you read the [contributing guidelines](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/CONTRIBUTING.md). -->

 <!-- Please label this pull request according to what type of issue you are addressing (see ../CONTRIBUTING.md) -->
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR reorganizes the disk pages into their own section and adds a new documentation section discussing disk encryption options.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
N/A

**Special notes for your reviewer**:
<!-- Refer to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/docs/book/src/developers/releasing.md#release-support for more information about which changes are eligible for backport -->

- [ ] cherry-pick candidate

**TODOs**:
<!-- Put an "X" character inside the brackets of each completed task. Some may be optional depending on the PR. -->

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
Reorganized the disk pages into their own section and added a new documentation section discussing disk encryption options.
```

bryan-cox commented:
Happy to squash those two commits into one. Just left them separate for now for peer review.

mboersma commented:
/ok-to-test

mboersma commented on docs/book/src/topics/disk-encryption.md:
```diff
@@ -0,0 +1,55 @@
+
+When enabled, Azure Disk Storage SSE encrypts data stored on Azure managed disks, i.e. OS and data disks. This option can be enabled using customer managed keys.
+
+Customer managed keys must be configured through a Disk Encryption Set (DES) resource. For more information on Azure Disk Storage SSE, please see this [link](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption).
```
```suggestion
Customer-managed keys must be configured through a Disk Encryption Set (DES) resource. For more information on Azure Disk Storage SSE, please see this [link](https://learn.microsoft.com/azure/virtual-machines/disk-encryption).
```
This is nitpicky, but I try to remove the `en-us/` part of doc URLs so the user can access a localized version depending on their language preferences.

bryan-cox replied:
That makes total sense. I didn't even think about that. I'll try to remember that for next time :)

mboersma commented on docs/book/src/topics/disk-encryption.md:
```diff
@@ -0,0 +1,55 @@
+Customer managed keys must be configured through a Disk Encryption Set (DES) resource. For more information on Azure Disk Storage SSE, please see this [link](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption).
+
+### Example with OS Disk using DES
+When using customer managed keys, you only need to provide the DES ID within the managedDisk spec. 
```
```suggestion
When using customer-managed keys, you only need to provide the DES ID within the managedDisk spec. 
```

mboersma reviewed (commented):
This is great @bryan-cox, thank you! I just had a few nitpicks.

bryan-cox commented:
@mboersma -Thanks for the feedback! I've addressed all the comments. Just let me know when you think it's good to go and I'll squash them all into one commit.

mboersma reviewed (commented):
/lgtm

Squash whenever time permits and I'm happy to approve this PR.

bryan-cox commented:
Commits are all squashed now.

mboersma reviewed (approved):
/approve
